---
title: C6320
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C6320
helpviewer_keywords:
- C6320
ms.assetid: fb9e568e-b3d4-4ce2-a276-a64ad74d7b1e
author: mikeblome
ms.author: mblome
manager: markl
ms.workload:
- multiple
ms.openlocfilehash: 8d5667f038974a5c814ab4e9e973274b0617fb26
ms.sourcegitcommit: 174c992ecdc868ecbf7d3cee654bbc2855aeb67d
ms.translationtype: MT
ms.contentlocale: it-IT
ms.lasthandoff: 12/06/2019
ms.locfileid: "74878021"
---
# <a name="c6320"></a>C6320
avviso C6320: l'espressione di filtro eccezioni è la costante EXCEPTION_EXECUTE_HANDLER. Questa condizione potrebbe mascherare eccezioni la cui gestione non è prevista

 Questo avviso indica l'effetto collaterale dell'utilizzo di EXCEPTION_EXECUTE_HANDLER costante nel blocco __except. In questo caso, l'istruzione nel blocco \__except viene sempre eseguita per gestire l'eccezione, incluse le eccezioni che non si desidera gestire in una funzione specifica. Si consiglia di controllare l'eccezione prima di gestirla.

## <a name="example"></a>Esempio
 Il codice seguente genera questo avviso perché il blocco __except tenterà di gestire le eccezioni di tutti i tipi:

```cpp
#include <stdio.h>
#include <excpt.h>

void f(int *p)
{
   __try
   {
      puts("in try");
      *p = 13;  // might cause access violation exception
      // code ...
   }
   __except(EXCEPTION_EXECUTE_HANDLER) // warning
   {
      puts("in except");
      // code ...
   }
}
```

 Per correggere il problema, utilizzare `GetExceptionCode` per verificare la presenza di una particolare eccezione prima di gestirla, come illustrato nel codice seguente:

```cpp
#include <stdio.h>
#include <windows.h>
#include <excpt.h>

void f(int *p)
{
   __try
   {
      puts("in try");
      *p = 13;    // might cause access violation exception
      // code ...
   }
   __except(GetExceptionCode()==EXCEPTION_ACCESS_VIOLATION ?
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)
   {
      puts("in except");
      // code ...
   }
}
```

## <a name="see-also"></a>Vedere anche
 [Istruzione try-except](/cpp/cpp/try-except-statement)
